portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,153 @@
+    return num;
+  }));
+}
+
+/**
+ * Very basic pagination feature...
+ * Needs to be refined to have an upper limit on number of page buttons
+ */
+function pagination() {
+  var numShown = getCommentLimit();
+  (fetch('/num-comments').then(response => response.text()).then(numComments =>
+  {
+    var numPages = Math.ceil(numComments/numShown);
+    var paginationDiv = document.getElementById("pagination");
+    for (var pageNumber = 0; pageNumber < numPages; pageNumber++) {
+      const thisPageNumber = pageNumber;
+
+      var pageElement = document.createElement("a");
+      pageElement.innerText = thisPageNumber;
+      pageElement.href = "#";
+      pageElement.addEventListener('click', () => {
+        //load comments on pageNumber
+        addMessage(thisPageNumber, getSortMethod);
+      });
+
+      paginationDiv.appendChild(pageElement);
+    }
+  }));
+}
+
+
+/**
+ * Using fetch request content from servlet and add to home page
+ */
+function addMessage(pageNumber = 0, sortMethod = "DESCENDING") {
+  /**
+   * GET does not support additional URL params in the body
+   * So...
+   * A novel hack around this (rather than rewrite the GET logic)
+   * Is to just manually add the strings into the fetch url
+   * This 'works' because the java function getParameter 
+   * Will still parse a manually constructed URL string
+   * So...
+   */
+  const numShown = getCommentLimit();
+  clearMessageDiv();
+  const url = `/data?pageNumber=${pageNumber}&numShown=${numShown}&sortMethod=${sortMethod}`;
+  fetch(url).then(response => response.json()).then((quote) => {
+    const messageContainerDiv = document.getElementById("message-container");
+    quote.forEach(Comment => messageContainerDiv.appendChild(createMessageElements(Comment)));
+  });
+}
+
+/**
+ * Build the elements of an individual message using JSON data
+ * Also create a delete button
+ */
+function createMessageElements(Comment) {
+  var messageDiv = document.createElement("div");
+
+  var commentElement = document.createElement("p");
+  commentElement.innerText = `${Comment.comment} posted at ${Comment.timestamp}`;
+
+  const deleteButtonElement = document.createElement("button");
+  deleteButtonElement.innerText = "Delete comment";
+  deleteButtonElement.addEventListener('click', () => {
+    //remove from datastore and DOM
+    deleteComment(Comment);
+    messageDiv.remove();
+  });
+
+  var likeAndButtonElements = document.createElement("div");
+  var likesElement = document.createElement("p");
+  var likes = 0;
+  likesElement.innerText = likes.toString() + " upvotes";
+
+  var likeButtonElement = document.createElement("button");
+  likeButtonElement.innerText = "Upvote";
+  likeButtonElement.addEventListener('click', () => {
+    //TODO: ADD BACKEND SUPPORT FOR THIS...
+    likes++;
+    likesElement.innerText = likes + " upvotes";
+  });
+
+  likeAndButtonElements.appendChild(likesElement);
+  likeAndButtonElements.appendChild(likeButtonElement);
+  
```

Comment:
If there's no back-end support for it, don't check it in.  Don't add to the front-end before you've implemented on the back-end.

Reply:
Noted for future; will leave in for now as an upstream branch requires these specifically and don't want to cause some headache down the line